src/design-system/InputButton.tsx

```diff
@@ -10,15 +14,21 @@ const StyledButton = styled.button`
   height: 48px;
   width: 200px;
   outline: none;
+  ${(props: StyleProps) => props.styles as InterpolationFunction<StyleProps>};
```

Comment:
this is clever but I also think I don't quite understand what's going on here. there's a bunch of assertions being made, how sure are you that this is type-safe?

Reply:
I am not sure! I'm still learning about Typescript 😄

I was trying to find a way to make this button's style more extendable without having to define every single property with a ternary expression, and when I first implemented this without the `as InterpolationFunction<StyleProps` I saw the following error:

```
Argument of type '(props: StyleProps) => React.CSSProperties' 
is not assignable to parameter of type 'Interpolation
 <ThemedStyledProps
 <Pick<DetailedHTMLProps
 <ButtonHTMLAttributes<HTMLButtonElement>, ...
```

Have you seen an error similar to that before? I assume it might mean that the interpolated `props` in this styled-component expects to be one of those types listed, instead of `React.CSSProperties`, but I wasn't sure. I researched until I found someone doing something similar and noticed that `as InterpolationFunction<Props>` example. 

Is there another way you've found to extend styled-components styles? Would it be preferable to include the style like this?

```
const StyledButton = styled.button<StyleProps>`
  background: #00615c;
  font-size: 16px;
  border-radius: 12px;
  color: white;
  font-family: "Poppins", sans-serif;
  height: 48px;
  width: ${props => props.width ? props.width : "200px"};
  outline: none;
`;
```